libs/sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -1991,10 +1991,16 @@ public Node visitBitwiseBinary(SqlBaseParser.BitwiseBinaryContext context) {
 
     @Override
     public Node visitConcatenation(SqlBaseParser.ConcatenationContext context) {
+        var left = (Expression) visit(context.left);
+        var right = (Expression) visit(context.right);
+        if (left instanceof NullLiteral && right instanceof NullLiteral) {
+            return NullLiteral.INSTANCE;
+        }
```

Comment:
The main problem here is that the `null || null -> null` operator behaves different that `concat(null, null)  -> empty string`. The only other solution I'd come up with is passing the information that this was original an operator somehow to the function implementation which sounds complex.